pkg/northbound/gnmi/set.go

```diff
@@ -527,6 +550,52 @@ func doRollback(changes mapNetworkChanges, mgr *manager.Manager, target string,
 		errResp.Error(), rolledbackIDs)
 }
 
+func listenAndBuildResponse(mgr *manager.Manager, changeID network.ID) ([]*gnmi.UpdateResult, error) {
+	networkChan := make(chan *network.NetworkChange)
+	errWatch := mgr.NetworkChangesStore.Watch(networkChan)
```

Comment:
It's a pity this Watch() can't be filtered by "changeID" . You could get a very big set of results here and you only want 1. If you could filter by "changeID" you might get max 3.